### PR TITLE
[zh-cn]: update the translation of `return` statement section

### PR DESCRIPTION
**Pull request by T34-active**
### Related issues and pull requests
https://github.com/mdn/translated-content/issues/18266

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/statements/return/index.md [hunk 1]
```diff
@@ -1,68 +1,54 @@
...
 
-```js
-function square(x) {
-  return x * x;
-}
-const demo = square(3);
-// demo will equal 9
-```
+- 在普通函数中，对该函数的调用会计算出返回值。
+- 在异步函数中，生成的 promise 将以返回的值解析。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/statements/return/index.md:29]: 参考 https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Promise

```suggestion
- 在异步函数中，生成的 promise 将以返回的值敲定。
```

File: files/zh-cn/web/javascript/reference/statements/return/index.md [hunk 2]
```diff
@@ -1,68 +1,54 @@
...
 ```
 
-- `expression`
+- `expression` {{optional_inline}}
   - : 要返回的值的表达式。如果忽略，则返回 `undefined`。
 
 ## 描述
 
-当在函数体中使用 `return` 语句时，函数将会停止执行。如果指定一个值，则该值会被返回给函数调用者。例如，以下函数返回其参数 `x` 的平方，其中 `x` 是数字。
+`return` 语句只能在函数体内使用。当在函数体中使用 `return` 语句时，函数的执行会停止。`return` 语句在不同类型的函数中具有不同的效果：
```
Comment (T34-active) [inline files/zh-cn/web/javascript/reference/statements/return/index.md:26]:
```suggestion
`return` 语句只能在函数体内使用。当 `return` 出现在函数体内部时，该函数的执行将会停止。`return` 语句在不同类型的函数中具有不同的效果：
```

File: files/zh-cn/web/javascript/reference/statements/return/index.md [hunk 3]
```diff
@@ -1,68 +1,54 @@
...
 
-```js
-function square(x) {
-  return x * x;
-}
-const demo = square(3);
-// demo will equal 9
-```
+- 在普通函数中，对该函数的调用会计算出返回值。
```
Comment (T34-active) [inline files/zh-cn/web/javascript/reference/statements/return/index.md:28]:
```suggestion
- 在普通函数中，对该函数的调用会返回其返回值。
```

File: files/zh-cn/web/javascript/reference/statements/return/index.md (hunk 1 above)
Comment (T34-active) [inline files/zh-cn/web/javascript/reference/statements/return/index.md:29]:
```suggestion
-  在异步函数中，产生的 promise 将以返回值完成（resolve）。
```

File: files/zh-cn/web/javascript/reference/statements/return/index.md [hunk 4]
```diff
@@ -1,68 +1,54 @@
...
+- 在异步函数中，生成的 promise 将以返回的值解析。
+- 在生成器函数中，生成的生成器对象的 `next()` 方法返回 `{ done: true, value: returnedValue }`。
```
Comment (T34-active) [inline files/zh-cn/web/javascript/reference/statements/return/index.md:30]:
```suggestion
- 在生成器函数中，生成的生成器对象的 `next()` 方法将返回 `{ done: true, value: returnedValue }`。
```

File: files/zh-cn/web/javascript/reference/statements/return/index.md [hunk 5]
```diff
@@ -1,68 +1,54 @@
...
-return x;
-return x + y / 3;
-```
+如果在 {{jsxref("Statements/try...catch", "try")}} 块内执行了 `return` 语句，它会首先执行 `finally` 块然后才实际返回值（如果存在）。
 
-### 自动插入分号
+### 自动分号补全
 
-[自动插入分号（ASI）](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar#自动分号补全)规则会影响 `return` 语句。在 `return` 关键字和被返回的表达式之间不允许使用换行符。
+语法禁止在 `return` 关键字和要返回的表达式之间存在换行符。
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/statements/return/index.md:37]:
```suggestion
语法禁止在 `return` 关键字和要返回的表达式之间使用换行符。
```

File: files/zh-cn/web/javascript/reference/statements/return/index.md [hunk 6]
```diff
@@ -1,68 +1,54 @@
...
 return;
 a + b;
 ```
 
-控制台会警告“unreachable code after return statement”。
-
-> **备注：** 从 Firefox 40 开始，如果在 `return` 语句后发现不可达的代码，控制台会显示一个警告。
-
-为了避免这个问题（防止 ASI），你可以使用括号：
+这会使函数返回 `undefined`，而 `a + b` 表达式永远不会被计算。这可能会在[控制台中生成警告](/zh-CN/docs/Web/JavaScript/Reference/Errors/Stmt_after_return)，为避免此问题（防止 ASI），你可以使用括号：
```
Comment (jasonren0403) [inline files/zh-cn/web/javascript/reference/statements/return/index.md:51]:
```suggestion
这会使函数返回 `undefined`，而 `a + b` 表达式永远不会被计算。这可能会在[控制台中生成警告](/zh-CN/docs/Web/JavaScript/Reference/Errors/Stmt_after_return)。

为避免此问题（防止 ASI），你可以使用括号：
```

File: files/zh-cn/web/javascript/reference/statements/return/index.md [hunk 7]
```diff
@@ -1,68 +1,54 @@
...
-function square(x) {
-  return x * x;
-}
-const demo = square(3);
-// demo will equal 9
-```
+- 在普通函数中，对该函数的调用会计算出返回值。
+- 在异步函数中，生成的 promise 将以返回的值解析。
+- 在生成器函数中，生成的生成器对象的 `next()` 方法返回 `{ done: true, value: returnedValue }`。
+- 在异步生成器函数中，生成的异步生成器对象的 `next()` 方法返回一个已兑现的 promise，其值为 `{ done: true, value: returnedValue }`。
```
Comment (T34-active) [inline files/zh-cn/web/javascript/reference/statements/return/index.md:31]:
```suggestion
- 在异步生成器函数中，生成的异步生成器对象的 `next()` 方法返回一个已兑现的 promise，其值为 `{ done: true, value: 返回的值}`。
```

**Review by jasonren0403 — COMMENTED** [review]
一些问题

**Review by jasonren0403 — APPROVED** [review]

**Comment by T34-active** [comment]
@jasonren0403 Thank you!

**Pull request merged**